lib/Basic/PlatformUtility.cpp

```diff
@@ -348,8 +348,14 @@ std::string sys::makeTmpDir() {
   CreateDirectoryW((LPCWSTR)wPath.data(), NULL);
   return std::string(path);
 #else
-  char tmpDirPathBuf[] = "/tmp/fileXXXXXX";
-  return std::string(mkdtemp(tmpDirPathBuf));
+  char *tmpDirPath = nullptr;
+  if (const char *tmpDir = std::getenv("TMPDIR"))
+    asprintf(&tmpDirPath, "%s/fileXXXXXX", tmpDir);
+  else
+    tmpDirPath = "/tmp/fileXXXXXX";
+  auto tmpDirString = std::string(mkdtemp(tmpDirPath));
+  free(tmpDirPath);
```

Comment:
🤔 This may now attempt to free the inline `"/tmp/fileXXXXXX"`.

Reply:
I think this last iteration should finally work fine.